castle/cms/skins/castle/login_form.cpt

```diff
@@ -60,7 +58,10 @@
                         target request/target|nothing;
                         target python:test(target in ('_parent', '_top', '_blank', '_self'), target, None);
                         ztu modules/ZTUtils;
-                        login nocall: context/@@secure-login">
+                        login nocall: context/@@secure-login;
+                        backend_urls python:context.portal_registry['plone.backend_url'] or [];
+                        only_allow_login_to_backend_urls python:context.portal_registry['plone.only_allow_login_to_backend_urls'];
+                        bad_domain python: only_allow_login_to_backend_urls and len(backend_urls) > 0 and portal_url.rstrip('/') not in backend_urls;">
```

Comment:
I know I wrote this expression originally but it's kinda ugly... is it possible to improve?